### ISSUE TITLE
Alexa broke JSON request parsing by sending timestamp in Unix time format in addition to ISO 8601

### ISSUE THREAD

**Issue by chmasi** [AreYouFreeBusy/AlexaSkillsKit.NET#37]
I have created a skill a couple of weeks ago which works pretty fine. Now it stoppt working. I did haven´t changend anything, my SSL Certificate is valid until 2020, so whats wrong? Did Amazon change something?

This is the message I am getting back:
There was an error calling the remote endpoint, which returned HTTP 400 : InvalidJson

Martin

**Comment by jaysonhelseth** [comment]
I am having the same issue with a skill that was working on Friday. July, 7, 2017.

Jayson

**Comment by stefann42** [comment]
If a skill using this library is responding with HTTP 400 : InvalidJson it means the Alexa service is sending something in the request payload that the library isn't aware of and unable to parse. If you can capture that payload and share it here it would be very helpful to troubleshoot the issue.

**Comment by amp64** [comment]
Amazon changed something that broke this: I am using these bits live in production, and its fine. If I point a development version at the EXACT SAME LIVE SERVICE, I get the json error too. Amazon have changed something such that skills deployed after a certain date (sometime less than a month ago) need to work subtly differently, somewhere. On the other hand skills deployed over a month ago continue to work just fine.

I can see from the Azure logs that I am returning a 400, but I cannot see why.

Is there any way to get logging out of the existing binary? Over the weekend I will build my own version and do some experiments, if no-one else gets the chance before then.

**Comment by amp64** [comment]
Here is what the HttpRequest contains:
![image](https://user-images.githubusercontent.com/20894022/28071896-2d701c00-6606-11e7-847d-fcf780ee49f8.png)
Sorry its a bitmap, but I can't figure out how to get it as proper text from App Insights.

**Comment by amp64** [comment]
From what I can tell, Speechlet.GetResponseAsync(request) is working, but that is being turned into a 400 before it gets out of the server. This matches what is described in Issue #32

**Comment by jaysonhelseth** [comment]
The Amazon test harness is sending the timestamp as a long, while the device is sending the timestamp as ISO 8601. Here is an ongoing post about it: https://forums.developer.amazon.com/questions/78372/timestamp-is-sent-in-different-formats.html?page=2&pageSize=10&sort=votes

I suppose then timestamp will have to be more fliexible if Amazon is going to change the rules.

**Comment by jaysonhelseth** [comment]
I opened a pull request  #38  that hopefully we get the ball rolling on the fix.

**Comment by EgMan** [comment]
I am getting the same issue.  Here is a json if it helps troubleshoot.  I removed "userID"
```
{  
   "session":{  
      "sessionId":"SessionId.710422a8-4f2b-4dd5-8a59-81995f7eb6d2",
      "application":{  
         "applicationId":"amzn1.ask.skill.cde8bde3-7629-4d89-9871-952be8243831"
      },
      "attributes":{  

      },
      "user":{  
         "userId":"I removed this field"
      },
      "new":true
   },
   "request":{  
      "type":"IntentRequest",
      "requestId":"EdwRequestId.295020ef-52a7-42c5-8d7e-ff6d103e9f13",
      "locale":"en-US",
      "timestamp":"2017-07-12T19:10:19Z",
      "intent":{  
         "name":"ClosestHospitalIntent",
         "slots":{  

         }
      }
   },
   "version":"1.0"
}
```

**Comment by jejernig** [comment]
Do you have any tests for time stamp checking @jaysonhelseth?

**Comment by jaysonhelseth** [comment]
I can try to add some unit tests for the code tomorrow (July 14, 2017). If I don't get to it then I won't be able to for about a week. Thanks for the reminder on the unit tests.

**Comment by jaysonhelseth** [comment]
The unit tests have been committed.

**Comment by jejernig** [comment]
I will be happy to help contribute. Mostly because I keep failing certification because of my end point not accepting valid signed requests.

**Comment by realares** [comment]
Amazon sends TimeStamps as String "2017-07-12T19:10:19Z" AND as UnixTimestamp like 34511252. This cause an Parsing Exception.. Some Forks fixed this Problem. Like mine :D https://github.com/realares/Ra.AlexaSkillsKit.NET

**Comment by mikelor** [comment]
Looks like a pull request was submitted by @jaysonhelseth, I was able to take his fork, with the Timezone check, gen up a new nuget, recompile my Skill and it now works again. What's the next step of getting this bumped up to 1.5.2? How can I help?

**Comment by stefann42** [comment]
This issue is now fixed in version 1.5.2.

I accepted pull request https://github.com/AreYouFreeBusy/AlexaSkillsKit.NET/pull/38 but reimplemented it because it had a few issues of its own (see PR comments).
